test/new_relic/newrelicyml_test.rb

```diff
@@ -0,0 +1,144 @@
+  end
+
+  def test_default_value_nil
+    assert_equal 'nil', NewRelicYML.default_value(:begonia, FAKE_CONFIGS[:begonia])
+  end
+
+  def test_sanitize_description_from_callouts
+    assert_equal '  White flowers.', NewRelicYML.sanitize_description(FAKE_CONFIGS[:lily][:description])
+  end
+
+  def test_sanitize_description_from_backticks
+    assert_equal 'If true, green with white polka dots.', NewRelicYML.sanitize_description(FAKE_CONFIGS[:begonia][:description])
+  end
+
+  def test_sanitize_description_from_hyperlinks
+    assert_equal 'Leafy and pretty.', NewRelicYML.sanitize_description(FAKE_CONFIGS[:monstera][:description])
+  end
+
+  def test_sanitize_description_from_popovers
+    assert_equal 'Low maintenance.', NewRelicYML.sanitize_description(FAKE_CONFIGS[:pothos][:description])
+  end
+
+  def test_format_description
+    assert_equal '  # White flowers.', NewRelicYML.format_description('White flowers.')
+  end
+
+  def test_get_configs
+    assert_equal final_config_hash, NewRelicYML.get_configs(FAKE_CONFIGS)
+  end
+
+  def test_build_string
+    assert_equal final_string, NewRelicYML.build_string(FAKE_CONFIGS)
+  end
+
+  private
+
+  FAKE_CONFIGS = {
+    :begonia => {
+      :default => nil,
+      :documentation_default => nil,
+      :public => true,
+      :description => 'If `true`, green with white polka dots.'
+    },
+    :lily => {
+      :default => 1,
+      :documentation_default => 2,
+      :public => true,
+      :description => <<~DESCRIPTION
+        <Callout variant="caution">
+          White flowers.
+        </Callout>
+      DESCRIPTION
+    },
+    :'monstera' => {
+      :default => '',
+      :public => true,
+      :description => 'Leafy and [pretty](/pretty/plants).'
+    },
+    :'pothos' => {
+      :default => '',
+      :public => false,
+      :description => 'Low <InlinePopover type="maintenance" />.'
+    },
+    :'app_name' => {
+      default: 'My Application',
+      :public => true,
+      :description => 'An app'
+    },
+    :'defer_rails_initialization' => {
+      :default => false,
+      :public => true,
+      :description => 'This should never be included'
+    },
+    :'config_path' => {
+      :default => 'newrelic.yml',
+      :public => true,
+      :description => 'Config path'
+    }
+  }
+
+  def final_config_hash
+    config_hash = {
+      :begonia => {
+        :description => '  # If true, green with white polka dots.',
+        :default => 'nil'
+      },
+      :config_path => {
+        :description => '  # Config path',
+        :default => 'newrelic.yml'
+      },
+      :lily => {
+        :description => '  # White flowers.',
+        :default => 2
+      },
+      :monstera => {
+        :description => '  # Leafy and pretty.',
+        :default => '""'
+      }
+    }
+
+    config_hash
+  end
+
+  # rubocop:disable Style/RedundantStringEscape
+  def final_string
+    <<~YML
+      \ \ # If true, green with white polka dots.
```

Comment:
Are the backslashes used to preserve spaces? If so, maybe a different style HEREDOC declaration would work that doesn't trim away the spaces.